projects/plugins/jetpack/extensions/blocks/recurring-payments/edit.jsx

```diff
@@ -15,17 +15,40 @@ import { getBlockStyles } from './util';
 const BLOCK_NAME = 'recurring-payments';
 
 export default function Edit( { attributes, clientId, setAttributes } ) {
-	const { align, planId, width } = attributes;
-	const planIds = useMemo( () => {
-		const _planIds = ( '' + planId ).split( '+' ).map( id => parseInt( id, 10 ) );
-		return _planIds;
-	}, [ planId ] );
+	const { align, planId, planIds, width } = attributes;
+
+	// planId is a integer, planIds is an array.
+	// if planIds is set, use it, otherwise use planId. Going forward we should only use planIds.
+	const _planIds = useMemo( () => {
+		return planIds || ( planId ? [ planId ] : [] );
+	}, [ planId, planIds ] );
+
 	const editorType = getEditorType();
 	const postLink = useSelect( select => select( editorStore )?.getCurrentPost()?.link, [] );
 
 	const updateSubscriptionPlans = useCallback(
 		newPlanIds => {
-			const newPlanId = newPlanIds !== planIds ? newPlanIds.join( '+' ) : planId;
+			// console.log( 'updateSubscriptionPlans', planIds, newPlanIds );
+
+			// verify _planIds is a non-empty array.
+			if ( ! Array.isArray( newPlanIds ) || 0 === newPlanIds.length ) {
+				return;
+			}
+			// ensure/convert all elements to integers.
+			const validatedPlanIds = newPlanIds
+				.map( id => parseInt( id, 10 ) )
+				.filter( id => ! isNaN( id ) );
+
+			// if all the elements match the existing planIds, do nothing.
+			if (
+				Array.isArray( planIds ) &&
+				validatedPlanIds.length === planIds.length &&
+				validatedPlanIds.every( i => planIds.includes( i ) )
+			) {
+				return;
+			}
+
+			const newPlanId = validatedPlanIds.join( '+' );
```

Comment:
Not sure what this is. `planId` is still an integer and I don't think this is necessary anyway.

Reply:
This is used to create the URL (in the function `resolvePaymentUrl()`).  I'm not sure it actually does much either but it creates a unique URL that gets saved with the block too.  I figured it was better to not mess with it too much.